src/components/button/Button.styles.ts

```diff
@@ -67,3 +68,15 @@ export const StyledTypography = styled(Typography)<StyledTypographyProps>`
   flex-grow: 1;
   text-align: ${({ $textAlign }) => $textAlign};
 `;
+
+interface StyledIconProps {
+  $color: ButtonColorTypes;
+  $variant: ButtonVariants;
+  $isLoading: boolean;
+  disabled: boolean;
+}
+
+export const StyledIcon = styled(Icon)<StyledIconProps>`
+  color: ${({ theme, $color, $variant, disabled, $isLoading }) =>
```

Comment:
Perhaps we can use the color prop from the Icon component and pass the color from src/components/button/Button.tsx calling getIconColor function